tests/test_crawl.py

```diff
@@ -1,8 +1,12 @@
 import json
 import socket
+import logging
+
+from testfixtures import LogCapture
```

Comment:
That's an interesting idea, but I'd prefer a dependency. Django maintains a backport of assertLogs, unittest2 maintains a backport of assertLogs, it'd be too much copy-paste if Scrapy start to do the same. Using unittest2 dependency instead of testfixtures sounds good to me - the code becomes more compatible with 3.x standard library. But it could be harder from technical point of view: Scrapy uses twisted.trial.unittest.TestCase which inherits from 'unittest.TestCase', not from 'unittest2.TestCase'.